File: mobile-wallet-master/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/presenter/ReadQrPresenter.java
package org.mifos.mobilewallet.mifospay.qr.presenter;

import org.mifos.mobilewallet.core.base.UseCaseHandler;
import org.mifos.mobilewallet.mifospay.base.BaseView;
import org.mifos.mobilewallet.mifospay.qr.QrContract;

import javax.inject.Inject;

/**
 * Created by naman on 7/9/17.
 */

public class ReadQrPresenter implements QrContract.ReadQrPresenter {

    private final UseCaseHandler mUsecaseHandler;
    private QrContract.ReadQrView mReadQrView;


    @Inject
    public ReadQrPresenter(UseCaseHandler useCaseHandler) {
        this.mUsecaseHandler = useCaseHandler;
    }

    @Override
    public void attachView(BaseView baseView) {
        this.mReadQrView = (QrContract.ReadQrView) baseView;
        mReadQrView.setPresenter(this);

    }

}

File: feature/qr/src/main/kotlin/org/mifospay/feature/read/qr/ReadQrPresenter.kt
package org.mifospay.feature.read.qr

import org.mifospay.core.data.base.UseCaseHandler
import org.mifospay.base.BaseView
import org.mifospay.qr.QrContract
import org.mifospay.qr.QrContract.ReadQrView
import javax.inject.Inject

/**
 * Created by naman on 7/9/17.
 */
class ReadQrPresenter @Inject constructor(private val mUsecaseHandler: UseCaseHandler) :
    QrContract.ReadQrPresenter {
    private var mReadQrView: ReadQrView? = null
    override fun attachView(baseView: BaseView<*>?) {
        mReadQrView = baseView as ReadQrView?
        mReadQrView!!.setPresenter(this)
    }
}